index.js

```diff
@@ -10,6 +10,20 @@ function isAgent(v) {
   return v && typeof v.addRequest === 'function';
 }
 
+function urlInstanceToObject(url) {
+  var urlObject = {};
+  // If the opts are genearted with the WHATWG API
+  if (url && Object.getOwnPropertySymbols && Object.getOwnPropertySymbols(url)[0]) {
+    var urlContext = url[Object.getOwnPropertySymbols(url)[0]];
+    for (var key of Object.keys(urlContext)) {
+      urlObject[key] = urlContext[key];
+    }
+  } else {
+    urlObject = Object.assign({}, url);
```

Comment:
@TooTallNate - are you okay dropping support for Node < 8 which would allow us to use object spread and maybe some more niceties?

Reply:
That is a great question!

Reply:
I'd like to keep Node 6 support at least.